### PR TITLE
Media & Text - Media picker buttons functionality

### PR DESCRIPTION
**Pull request by geriux**
Gutenberg PR https://github.com/WordPress/gutenberg/pull/17537

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1313

To test check https://github.com/WordPress/gutenberg/pull/17537 for a full description and screenshots.

It also includes a small change in `RNReactNativeGutenbergBridge.swift` to pass the media type.

#### Parent apps PRs

**WPiOS**: https://github.com/wordpress-mobile/WordPress-iOS/pull/12581
**WPAndroid**: https://github.com/wordpress-mobile/WordPress-Android/pull/10567 (it has the changes from the multiple media PR) but there aren't any new native changes, just the updated gutenberg reference

---

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by koke** [comment]
I tried to upload a video and after the upload finished, I got this warning:

> Failed to get size for image: https://sandbox.koke.me/wp-content/uploads/2019/09/img_1624.mov

And the video part in media-text only shows the placeholder.

**Comment by geriux** [comment]
> I tried to upload a video and after the upload finished, I got this warning:
> 
> > Failed to get size for image: https://sandbox.koke.me/wp-content/uploads/2019/09/img_1624.mov
> 
> And the video part in media-text only shows the placeholder.

Oh no, that's weird, do you know if I have to do something for the change of `ios/RNReactNativeGutenbergBridge.swift` to work? Did you test it locally? If so, you would have to rebuild because of the native change.

**Comment by koke** [comment]
> If so, you would have to rebuild because of the native change

🤦‍♂ My bad, I didn't realize there were native changes and was testing with the wrong binaries

**Comment by koke** [comment]
I got a weird issue... after testing the video upload, I removed the media-text block, inserted a new one and tried to upload an image. The media-text block will upload correctly but then try to show the image as a video and fail. When I closed the editor I got a crash:

```
Unhandled JS Exception: Cannot read property 'props' of null, stack:
<unknown>@177709:33
<unknown>@40386:21
_callTimer@40303:9
Object.callTimers@40510:9
MessageQueue.__callFunction@15816:44
<unknown>@15558:17
MessageQueue.__guard@15770:13
MessageQueue.callFunctionReturnFlushedQueue@15557:14
<unknown>@80:58
```

I haven't been able to reproduce the crash again 😞

**Comment by koke** [comment]
> <unknown>@177709:33

Looking at the generated JS, the crash seems to be in [KeyboardAwareFlatList](https://github.com/WordPress/gutenberg/blob/a56daebec30a3740116e7e495d9776ef4c3d51a6/packages/components/src/mobile/keyboard-aware-flat-list/index.ios.js#L33), so I would assume it's a coincidence unrelated to these changes, but let me know if you think you changed something that might be causing that.

**Comment by geriux** [comment]
> > @177709:33
> 
> Looking at the generated JS, the crash seems to be in [KeyboardAwareFlatList](https://github.com/WordPress/gutenberg/blob/a56daebec30a3740116e7e495d9776ef4c3d51a6/packages/components/src/mobile/keyboard-aware-flat-list/index.ios.js#L33), so I would assume it's a coincidence unrelated to these changes, but let me know if you think you changed something that might be causing that.

Not really, but I will double check just in case

**Comment by etoledom** [comment]
Milestone moved to [1.15](https://github.com/wordpress-mobile/gutenberg-mobile/milestone/24)

File: react-native-gutenberg-bridge/ios/GutenbergBridgeDelegate.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-public typealias MediaPickerDidPickMediaCallback = (_ media: [(Int32?,String?)]?) -> Void
+public typealias MediaPickerDidPickMediaCallback = (_ media: [(Int32?,String?,String?)]?) -> Void
```
Comment (etoledom) [inline react-native-gutenberg-bridge/ios/GutenbergBridgeDelegate.swift]: These callback parameter list is kind of growing a lot 🤔 . Maybe we could think about using a struct?
Having too many unnamed parameters in the public API doesn't sound like a good practice.
cc @koke @pinarol

Comment (koke) [reply react-native-gutenberg-bridge/ios/GutenbergBridgeDelegate.swift]: That sounds like a good idea to me 😄

**Pull request merged**